lib/firmata.js

```diff
@@ -1229,10 +1231,19 @@ Board.prototype._sendOneWireRequest = function(pin, subcommand, device, numBytes
 
 Board.prototype.setSamplingInterval = function(interval) {
   var safeint = interval < 10 ? 10 : (interval > 65535 ? 65535 : interval); // constrained
+  this.settings.samplingInterval = safeint;
   this.transport.write(new Buffer([START_SYSEX, SAMPLING_INTERVAL, (safeint & 0xFF), ((safeint >> 8) & 0xFF), END_SYSEX]));
 };
 
 /**
+ * Get sampling interval in millis. Default is 19 ms
+ */
+
+Board.prototype.getSamplingInterval = function(interval) {
+  return this.settings.samplingInterval;
```

Comment:
I should probably add a way to query the sampling interval in Firmata. I'm not sure about J5 / node-firmata, but some Firmata client's enable multiple connections to a single board (more common when using networking transports rather than serial). In this case, one client could set the sampling interval but another client would not know it has changed.